### PR TITLE
APIs

### PR DESCRIPTION
**Pull request by DyaryRaoof**
- Added function to create a game on the server via API when the page loads.
- Added a function to send new leader boards to the server via API.
- Added a function to save game Id in local storage.
- Added a function to fetch all game data from the server.
- Added a function to render all fetched games on the screen.
- Added a modal that displays when new games are successfully created on the server.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,18 +1,112 @@
...
+async function refreshLeaderboard() {
+  const gameId = retreiveGameId();
+  const endpoint = `games/${gameId}/scores/`;
+  const result = await fetch(URL + endpoint);
+  const json = await result.json();
+  const games = json.result;
+  renderScoreElements(games);
+}
+
+async function createGameAndSaveItsId() {
```
Comment (Dipeshtwis) [inline src/index.js]: - [ ] I would recommend using `arrow functions` instead of function keyword in all the places.

File: src/index.js [hunk 2]
```diff
@@ -1,18 +1,112 @@
...
+function saveGameIdLocally(id) {
+  localStorage.setItem('gameId', JSON.stringify(id));
+}
+
+function retreiveGameId() {
+  const id = JSON.parse(localStorage.getItem('gameId'));
+  return id;
+}
+
+function createScoreElement(userName, score) {
```
Comment (Dipeshtwis) [inline src/index.js]: - [ ] I would recommend using `arrow functions` instead of function keyword in all the places.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @DyaryRaoof, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: The app is working fine.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BrianSammit — APPROVED** [review]
# STATUS: APROVED :green_circle: :clap: 

Hello  @DyaryRaoof,
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**